101-hdinsight-custom-ambari-db/azuredeploy.json

```diff
@@ -0,0 +1,276 @@
+        "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+      }
+    },
+    "existingClusterStorageResourceGroup": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The resource group name of the storage account to use as the cluster's default storage."
+      }
+    },
+    "existingClusterStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the storage account to use as the cluster's default storage."
+      }
+    },
+    "newOrExistingClusterStorageContainerName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the storage container to use."
+      }
+    },
+    "existingSQLServerResourceGroup": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The name of the resource group where the existing SQL server is to use for the new external metastore SQL db."
+      }
+    },
+    "existingSQLServerName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the existing SQL server to use for the new external metastore SQL db."
+      }
+    },
+    "existingSQLServerUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The external Hive metastore's existing SQL server admin username."
+      }
+    },
+    "existingSQLServerPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The external Hive metastore's existing SQL server admin password."
+      }
+    },
+    "newMetastoreDBName": {
+      "type": "string",
+      "defaultValue": "[concat('metastoredb', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "The name of the new SQL db to create to serve as the external metastores."
+      }
+    },
+    "existingVirtualNetworkResourceGroup": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The existing virtual network resource group name."
+      }
+    },
+    "existingVirtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "The existing virtual network name."
+      }
+    },
+    "existingVirtualNetworkSubnetName": {
+      "type": "string",
+      "metadata": {
+        "description": "The existing virtual network subnet name."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": { 
+  	"sqldbname": "[concat(parameters('existingSQLServerName'), '/', parameters('newMetastoreDBName'))]",
+    "nestedtemplate_name": "[concat('nestedtemplate',uniqueString(resourceGroup().id))]"
```

Comment:
simplify this and just use a name that makes sense for the workload/deployment, eg. sqlDbDeployment

Reply:
Thanks. Fixed in line 144.